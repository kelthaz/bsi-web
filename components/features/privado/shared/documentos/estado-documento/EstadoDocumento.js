import React from 'react';
import PropTypes from 'prop-types';
import SvgAprobado from '../../../../../svgs/estados/SvgAprobado';

const EstadoDocumento = ({ label, documento, estadoDocumento }) => {
  const iconoEstado = (estado) => {
    switch (estado) {
      case 0:
        return (
          <button className="btn-mini-secondary" type="button">
            Subir
          </button>
        );
      case 1:
        return (
          <button className="btn-mini-secondary" type="button">
            Revisar
          </button>
        );
      case 2:
        return <SvgAprobado />;
      default:
        return <></>;
    }
  };

  return (
    <div className="row pb-3">
      <div className="col-5">
        <span className="body2">{label}</span>
      </div>
      <div className="col-5 text-overflow">
        <span className="link">{documento}</span>
      </div>
      <div className="col-2 text-center">{iconoEstado(estadoDocumento)}</div>
    </div>
  );
};

EstadoDocumento.propTypes = {
  label: PropTypes.string.isRequired,
  documento: PropTypes.string.isRequired,
  estadoDocumento: PropTypes.number.isRequired,
};

export default EstadoDocumento;
