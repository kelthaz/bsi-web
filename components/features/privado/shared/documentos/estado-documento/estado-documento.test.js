import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import * as React from 'react';
import EstadoDocumento from './EstadoDocumento';

describe('Pruebas en el componente EstadoDocumento', () => {
  const onClick = jest.fn();
  const click = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const props = { label: 'Acta constitutiva', documento: 'acta.pdf', estadoDocumento: 9 };
    // act
    const wrapper = shallow(<EstadoDocumento {...props} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrarse el estado aprobado', () => {
    // arrange
    const props = { label: 'Acta constitutiva', documento: 'acta.pdf', estadoDocumento: 2 };
    const wrapper = shallow(<EstadoDocumento {...props} />);
    // act
    const svgAprobado = wrapper.find('SvgAprobado');
    // assert
    expect(svgAprobado.exists()).toBe(true);
  });

  test('Debe de mostrarse el estado pendiente', () => {
    // arrange
    const props = { label: 'Acta constitutiva', documento: 'acta.pdf', estadoDocumento: 3 };
    const wrapper = shallow(<EstadoDocumento {...props} />);
    // act
    const svgPendiente = wrapper.find('SvgPendiente');
    // assert
    expect(svgPendiente.exists()).toBe(true);
  });

  test('Debe de mostrarse el estado rechazado y se debe llamar la funci贸n de cambiar', () => {
    // arrange
    const useRefSpy = jest.spyOn(React, 'useRef').mockReturnValueOnce({ current: { click } });
    const props = { label: 'Acta constitutiva', documento: 'acta.pdf', estadoDocumento: 4, onClick };
    const wrapper = shallow(<EstadoDocumento {...props} />);
    // act
    const svgRechazado = wrapper.find('SvgRechazado');
    wrapper.find('button').simulate('click');
    // assert
    expect(svgRechazado.exists()).toBe(true);
    expect(useRefSpy).toBeCalledTimes(1);
    expect(click).toBeCalledTimes(1);
  });

  test('Debe de mostrarse un boton con el label enviado y se debera llamar la funci贸n enviada', () => {
    // arrange
    const props = {
      label: 'Acta constitutiva',
      documento: 'acta.pdf',
      estadoDocumento: 1,
      labelButton: 'Firmar',
      onClick,
    };
    const wrapper = shallow(<EstadoDocumento {...props} />);
    // act
    const buttonSubir = wrapper.find('button');
    buttonSubir.simulate('click');
    // assert
    expect(buttonSubir.text()).toBe(props.labelButton);
    expect(onClick).toBeCalledTimes(1);
  });

  test('Debe de mostrarse un boton el boton de subir archivo', () => {
    // arrange
    const useRefSpy = jest.spyOn(React, 'useRef').mockReturnValueOnce({ current: { click } });
    const props = { label: 'Acta constitutiva', documento: 'acta.pdf', estadoDocumento: 0, onClick };
    const wrapper = shallow(<EstadoDocumento {...props} />);
    // act
    const buttonSubir = wrapper.find('button');
    buttonSubir.simulate('click');
    // assert
    expect(buttonSubir.text()).toBe('Subir');
    expect(useRefSpy).toBeCalledTimes(1);
    expect(click).toBeCalledTimes(1);
  });

  test('Debe de llamar la funci贸n enviada cuando se cambiar de archivo', () => {
    // arrange
    const props = { label: 'Acta constitutiva', documento: 'acta.pdf', estadoDocumento: 0, onClick };
    const wrapper = shallow(<EstadoDocumento {...props} />);
    const fileInput = wrapper.find('input');
    const file = new Blob(['file contents'], { type: 'text/plain' });
    // act
    fileInput.simulate('change', { target: { files: [file] } });
    // assert
    expect(onClick).toBeCalledTimes(1);
    expect(onClick).toBeCalledWith(file);
  });

  test('Debe de llamar la funci贸n enviada cuando se cambiar de archivo', () => {
    // arrange
    const props = { label: 'Acta constitutiva', documento: 'acta.pdf', estadoDocumento: 0, onClick };
    const wrapper = shallow(<EstadoDocumento {...props} />);
    const fileInput = wrapper.find('input');
    const file = new Blob(['file contents'], { type: 'text/plain' });
    // act
    fileInput.simulate('change', { target: { files: [file] } });
    // assert
    expect(onClick).toBeCalledTimes(1);
    expect(onClick).toBeCalledWith(file);
  });
});
