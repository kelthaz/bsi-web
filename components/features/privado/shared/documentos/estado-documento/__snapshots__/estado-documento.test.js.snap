// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente EstadoDocumento Debe de mostrarse correctamente el componente 1`] = `
<div
  className="row pb-3"
>
  <div
    className="col-5"
  >
    <span
      className="body2"
    >
      Acta constitutiva
    </span>
  </div>
  <div
    className="col-5 text-overflow"
  >
    <span
      className="link"
    >
      acta.pdf
    </span>
  </div>
  <div
    className="col-2 text-center"
  />
</div>
`;
